mautic/mautic
# Segment membership as a new filter for dynamic email content

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Add ability to use segment filters in email dynamic content.

#### Steps to test this PR:

1. Create a contact with email address `contact@a.email`
2. Create another contact with email address `contact@b.email`
3. Create "Segment A" with no filters
4. Create "Segment B" with no filters
5. Add the first contact to Segment A and the second contact to Segment B
7. Create email in `Channels > Emails`
    - Click on segment email
    - Choose both segments
    - Click builder
    - Drag dynamic content to email content 
    - Customize slot content with some default text
![Snímek obrazovky 2020-10-29 v 12 55 38](https://user-images.githubusercontent.com/12815758/97565518-715c9f00-19e6-11eb-8ee9-022953e6e205.png)
    - Click green `Add variant button` - and `Variation 1` will be added
    - Fill custom content for segment contacts
    - select Segment Membership 
 ![Snímek obrazovky 2020-10-29 v 12 47 23](https://user-images.githubusercontent.com/12815758/97565718-80435180-19e6-11eb-9996-2a14e7277c2d.png)
    - Save the email and send
    - The contact from Segment A will have content from Variant 1 and the other will see the default content
